File: csharp/Cheese.cs
// csharp/Cheese.cs
using System;

namespace ECommerce
{
    public class Cheese : ExpirableProduct, IShippable
    {
        public double Weight { get; protected set; }

        public Cheese(string name, double price, int quantity, DateTime expiryDate, double weight)
            : base(name, price, quantity, expiryDate)
        {
            Weight = weight;
        }

        public override bool RequiresShipping()
        {
            return true;
        }
    }
}

File: cpp/Cheese.cpp
#include "Cheese.h"

Cheese::Cheese(std::string name, double price, int quantity, std::chrono::system_clock::time_point expiryDate, double weight)
    : ExpirableProduct(name, price, quantity, expiryDate), weight(weight) {}

double Cheese::getWeight() const {
    return weight;
}

bool Cheese::requiresShipping() const {
    return true;
}
